llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -1446,6 +1448,92 @@ static bool isConditionTrueViaVFAndUF(VPValue *Cond, VPlan &Plan,
   return SE.isKnownPredicate(CmpInst::ICMP_EQ, VectorTripCount, C);
 }
 
+static bool useWideActiveLaneMask(VPlan &Plan, ElementCount VF, unsigned UF) {
+  VPRegionBlock *VectorRegion = Plan.getVectorLoopRegion();
+  VPBasicBlock *ExitingVPBB = VectorRegion->getExitingBasicBlock();
+  auto *Term = &ExitingVPBB->back();
+
+  using namespace llvm::VPlanPatternMatch;
+  if (!EnableWideActiveLaneMask || !VF.isVector() || UF == 1 ||
+      !match(Term, m_BranchOnCond(m_Not(m_ActiveLaneMask(
```

Comment:
The description of the patch says
> when it is safe to do so (i.e. using tail folding without runtime checks).

Is checking for ActiveLaneMask here enough to ensure that the unsafe cases are rejected?